plugins/Hosting/js/sagas/saga.js

```diff
@@ -204,6 +206,33 @@ function *requestDefaultSettingsSaga() {
 	}
 }
 
+// getHostStatus returns a promise that resolves with `true` if the provided
+// netaddress is connectable, or `false` if it is not connectable. The input
+// 'hostAddr' must be an address in the URL format, eg:
+// tcp://123.456.789.0:9982
+const getHostStatus = (hostAddr) => new Promise((resolve) => {
+	const hostUrl = url.parse(hostAddr)
+	const socket = net.connect(hostUrl.port, hostUrl.hostname)
+	socket.setTimeout(15e3) // 15 second host connectability timeout
```

Comment:
Certainly a less friendly user experience, though. You would need to save the initial values on startup and then check for an increase in non-error calls. Not sure how long it takes on average to get one connection.
I think having both would be ideal, but make it clear that one of them just means your port is properly opened. As an example, Transmission has a button that you can use to manually test your port:
![](http://www.techsupportalert.com/files/images/pc_freeware/free_online_books/optimize-transmission-windows-transmission-qt-windows-preferences-network-port-number.jpg)